driver-core/src/main/com/mongodb/BasicDBObject.java

```diff
@@ -187,6 +195,37 @@ public String toJson(final JsonWriterSettings writerSettings, final Encoder<Basi
         return writer.getWriter().toString();
     }
 
+    @Override
```

Comment:
Overriding this method, because the alternative is to add a new method to BasicBSONObject

     protected BSONEncoder getEncoder()

and override it here to return an instance of DefaultDBEncoder (which can handle DBRef), which I'd prefer not to pull into driver-core.

Reply:
Ack - thats fine.